src/modules/Snapper.rb

```diff
@@ -23,14 +23,119 @@
 # Summary:	Snapper settings, input and output functions
 # Authors:	Jiri Suchomel <jsuchome@suse.cz>
 #
-# $Id$
-#
 # Representation of the configuration of snapper.
 # Input and output routines.
+
 require "yast"
 
 module Yast
+
+
+  class Tree
```

Comment:
The Tree class is likely not so reusable since other users might want to save there data that a status with each node. In C++ templates would make the class reusable - in Ruby I don't know.

I tried to place the Tree class in a module Snapper but since Snapper is assigned SnapperClass.new this does not work. I leave that as room for improvement to ruby experts.